src/resmom/vnode_storage.c

```diff
@@ -162,23 +163,16 @@ add_vmapent_byID(void *ctx, const char *vnid, void *data)
 		(void) strncpy(pe->key, vnid, PBS_MAXNODENAME);
 		pe->recptr = data;
 		if (avl_add_key(pe, ctx) == AVL_IX_OK) {
-#ifdef	DEBUG
-			(void) sprintf(log_buffer, "avl_add_key IX_OK");
-			log_event(PBSEVENT_DEBUG, 0, LOG_DEBUG, __func__, log_buffer);
-#endif	/* DEBUG */
-			free (pe);
-			return (1);
-		} else {
-#ifdef	DEBUG
-			(void) sprintf(log_buffer, "avl_add_key not IX_OK");
-			log_event(PBSEVENT_DEBUG, 0, LOG_DEBUG, __func__, log_buffer);
-#endif	/* DEBUG */
-			free (pe);
+			free(pe);
 			return (0);
+		} else {
+			log_event(PBSEVENT_DEBUG, 0, LOG_DEBUG, __func__, "avl_add_key failed");
+			free(pe);
+			return (1);
 		}
 	} else
 		log_err(errno, __func__, "malloc pe failed");
 
-	return (0);
```

Comment:
According to the function header, in case of failure, we return 1. We will reach line 176 only in case of error in malloc. 
add_vmapent_byID() function is called in mom_vnode.c in add_mominfo() 
`	if (add_vmapent_byID(ctx, vnid, data) == 0)
		return (0);`
add_mominfo() returns 0 in case of success according to the function header. So I changed the return code to 1 in add_vmapent_byID(). Does that seem correct?